patches/api/0440-Allow-proper-checking-of-empty-item-stacks.patch

```diff
@@ -0,0 +1,81 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Aeltumn <daniel@goossens.ch>
+Date: Mon, 28 Aug 2023 13:41:09 +0200
+Subject: [PATCH] Allow proper checking of empty item stacks
+
+This adds a method to check if an item stack is empty or not. This mirrors vanilla's implementation of the same method.
+
+Also fixes incorrect nullability on getCursor().
+
+diff --git a/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java b/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
+index fe58058f9b5d29388d48115cc81dc48ab08c58c1..ac67a4e321f43d1ede09dafe2daa1f07de8a923f 100644
+--- a/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
++++ b/src/main/java/org/bukkit/event/inventory/InventoryClickEvent.java
+@@ -85,7 +85,7 @@ public class InventoryClickEvent extends InventoryInteractEvent {
+      *
+      * @return the cursor ItemStack
+      */
+-    @Nullable
++    @NotNull // Paper - fix nullability
```

Comment:
I think these two nullability fixes are a better fit for this patch: https://github.com/PaperMC/Paper/blob/b4e3b3d1dd447bac4cbf478595c1ec320bc6dd4b/patches/api/0172-Fix-Spigot-annotation-mistakes.patch